components/component.go

```diff
@@ -21,9 +20,13 @@ type Component struct {
 	// Add any other common fields across components below
 }
 
+func (c *Component) IsEnabled() bool {
+	return c.ManagementState == operatorv1.Managed
+}
+
 type ComponentInterface interface {
-	ReconcileComponent(owner metav1.Object, client client.Client, scheme *runtime.Scheme,
-		managementState operatorv1.ManagementState, namespace string, manifestsUri string) error
+	ReconcileComponent(cli client.Client, owner metav1.Object, namespace, manifestsUri string) error
+	IsEnabled() bool
```

Comment:
Following OpenShift best practices, we are moving to a 3-state management option for components (Managed, Unmanaged or Removed), so I am not sure introducing an IsEnabled() method into the interface is a good idea. I would suggest adding a ManagementState() method instead.

Reply:
`ManagedState` is an exported field of the base `Component` struct, so I am not sure if we really need to expose it through "getter". `IsEnabled` essentially encapsulates `managementState == operatorv1.Managed`. But perhaps we can entirely remove this and just inline this check in every component.

Reply:
Alternatively, we can just leave it as a `func` for the base `Component` which we anyway use to compose new structs but not enforce it as a contract for the interface. WDYT?

Reply:
`IsEnabled` essentially encapsulates `managementState == operatorv1.Managed` for convenience. Let me have a look.

Reply:
This is now addressed in f20cc74. I also rebased on top of latest `main`